assets/js/amazon-wc-checkout.js

```diff
@@ -79,6 +79,51 @@
 			renderButton( '#pay_with_amazon_cart', 'cart' );
 		}
 
+		/* Handles Amazon Pay Button on Product Pages. */
+		if ( $( '#pay_with_amazon_product' ).length > 0 ) {
+			var amazonProductBtn = renderButton( '#pay_with_amazon_product', 'product' );
+			if ( null !== amazonProductBtn ) {
+				amazonProductBtn.onClick( function() {
+					if ( $( '.single_add_to_cart_button' ).hasClass( 'disabled' ) ) {
+						$( '.single_add_to_cart_button' ).trigger( 'click' );
+						return;
+					}
+					var elemToBlock = $( '#pay_with_amazon_product' ).closest( 'div.summary' ).length > 0 ? $( '#pay_with_amazon_product' ).closest( 'div.summary' ) : false;
```

Comment:
Avoid running jQuery fn's multiple times. Better assign to a var once and check on that.

Reply:
Done!